### PR TITLE
support Linkvertise

### PR DESCRIPTION
**Pull request by 8MiYile**
Solve #9

### CODE REVIEW

File: plugin.php [hunk 1]
```diff
@@ -35,21 +29,41 @@ function check_for_redirect( $args ) {
...
-	$last_word = array_pop($pieces); //  get the keyword - this may not work if you use a plugin to allow slashes in your shortened url
-	return '//' . $_SERVER['SERVER_NAME'] . '/' . $last_word; // replace the '/' after $_SERVER['SERVER_NAME' if your yourls is not in your base domain, such as '/shorten/'
+  $protocol = (isset($_SERVER['HTTPS']) && $_SERVER['HTTPS'] === 'on' ? "https" : "http") ;
+  $actual_link = $protocol . "://$_SERVER[HTTP_HOST]$_SERVER[REQUEST_URI]"; // get the current url
+  $pieces = explode('/', $actual_link); // split the url into an arrray seperated by /
+  $last_word = array_pop($pieces); //  get the keyword - this may not work if you use a plugin to allow slashes in your shortened url
+  return $protocol . '://' . $_SERVER['SERVER_NAME'] . '/' . $last_word; // replace the '/' after $_SERVER['SERVER_NAME' if your yourls is not in your base domain, such as '/shorten/'
+}
+// About Linkvertise
+function btoa($str) {
```
Comment (HeroCC) [inline plugin.php]: Could we use the default [`base64_encode()`](https://www.php.net/manual/en/function.base64-encode.php) function instead?

File: plugin.php [hunk 2]
```diff
@@ -35,21 +29,41 @@ function check_for_redirect( $args ) {
...
+    $redirectUrl = getRedirect();
+    switch ( redirectService ) { 
+      case 'f': // Use adfocus
+        return ADFOCUS_DOMAIN . '/serve/sitelinks/?id=' . ADFOCUS_ID . '&url=' . $redirectUrl;
+      case 'a': // Adfly
+        return ADFLY_DOMAIN . '/' . ADFLY_ID . '/' . $redirectUrl;
+      case 'o': // OUO.io
+        return OUO_DOMAIN . '/qs/' . OUO_ID . '?s=' . $redirectUrl;
+      case 'l': // linkvertise.com
+        return getLinkvertise(LINKVERTISE_ID,$redirectUrl);
```
Comment (HeroCC) [inline plugin.php]: Missing a space here

File: README.md [hunk 3]
```diff
@@ -1,41 +1,4 @@
-# Conditional URL Advertisements
```
Comment (HeroCC) [inline README.md:1]: I'm not super comfortable having multiple translations in here since I'm not a native speaker and can't evaluate / vouch for the contribution. Could you change it back to just the one README file?

File: plugin.php [hunk 4]
```diff
@@ -3,22 +3,16 @@
...
+define( 'TRIGGERS', array('a/', 'f/', 'o/', 'l/') ); // Add any possible trigger to use here
 
-define( 'ADFLY_ID', '2777408' ); // Replace this with your Adfly ID
-define( 'ADFOCUS_ID', '287608' ); // Replace this with your Adfoc.us ID
-define( 'OUO_ID', '0IqYvHOo' ); // You get the drill
-
-define( 'ADFLY_DOMAIN', 'https://adf.ly' ); // If you have a custom Adfly domain, replace this with it
-define( 'ADFOCUS_DOMAIN', 'https://adfoc.us' ); // Same for this
-define( 'OUO_DOMAIN', 'https://ouo.io' ); 
+include 'user-config.php';
```
Comment (HeroCC) [inline plugin.php]: This is a good change too. If you'd like to make it another PR, that would be nice so it comes in as another commit rather than the single merge here.

**Review by HeroCC — CHANGES_REQUESTED** [review]
Hey! Looks good functionally, just a few other misc changes. I made comments throughout.

**Comment by 8MiYile** [comment]
Are you still there?
I thought you had left your project

File: plugin.php (hunk 4 above)
Comment (8MiYile) [reply plugin.php]: Do you mean this update is placed in another pr?
If this is the case, I can restore this part of the change

File: README.md (hunk 3 above)
Comment (8MiYile) [reply README.md:1]: As I said before, I can take care of the Chinese side

File: plugin.php (hunk 2 above)
Comment (8MiYile) [reply plugin.php]: In which line?
```php
return getLinkvertise(LINKVERTISE_ID, $redirectUrl);
```
look like this?

File: plugin.php (hunk 1 above)
Comment (8MiYile) [reply plugin.php]: I have actually included in this`base64_encode();`
I don't know why you would ask that

This part I remember I have in Readme explained the source, I am directly look at his writing, transfer to PHP, but also incidentally do a little optimization

File: plugin.php (hunk 2 above)
Comment (HeroCC) [reply plugin.php]: Correct

File: plugin.php (hunk 1 above)
Comment (HeroCC) [reply plugin.php]: I meant moreso why do we need the buffer conversion here? Can we not just use `base64_encode` directly in `getLinkvertise`?  I was under the impression `btoa` in JavaScript functions the same as `base64_encode` in PHP, does it not?

File: plugin.php (hunk 4 above)
Comment (HeroCC) [reply plugin.php]: Correct, feel free to move this part to another PR.

**Comment by HeroCC** [comment]
> Are you still there? I thought you had left your project

I'm still around. The plugin is generally stable and I don't use YOURLS daily anymore, so I don't make many updates here. But, I still am happy to accept PRs and do some maintenance in my free time.

File: plugin.php (hunk 1 above)
Comment (8MiYile) [reply plugin.php]: ~Okay, I'll try it, but I'm not sure if it will still pass the verification.~
Validated, pushing soon

**Comment by 8MiYile** [comment]
Okay, please recheck. If the changes pass, I'll push the user configuration piece along with it

**Comment by 8MiYile** [comment]
Next Update is 
image
